src/main/ruby/core/thread.rb

```diff
@@ -354,8 +353,32 @@ def thread_variables
     __thread_local_variables_lock { __thread_local_variables.keys }
   end
 
-  def self.start(*args, &block)
-    Thread.new(*args, &block)
+  class << self
+    def new(*args, &block)
+      thread = Truffle.invoke_primitive(:thread_allocate, self)
+      thread.send(:initialize, *args, &block)
```

Comment:
When there is a *splat or a &block I tend to use parens.
But I am fine removing them if you think it's better.